packages/app/search/service.ts

```diff
@@ -25,15 +26,28 @@ export async function search(
             throw modelError
         }
 
-        const searchResults = await searchDb.search(
+        const [{ metadata, results }] = await searchDb.search(
             parsedInput.modelName,
             titleProperty,
             parsedInput.query,
             parsedInput.resultsPerPage,
             parsedInput.pageNumber
         )
+        const resultsCount = metadata.length ? metadata[0]?.resultsCount : 0
 
-        return [null, searchResults]
+        return [
+            null,
+            {
+                metadata: {
+                    pageCount: Math.max(Math.ceil(resultsCount / resultsPerPage), 1),
```

Comment:
Zod's defaults / parsing prevent us from diving by zero or passing a number less than one. We a minimum page number of "1", so if we have 7 results we'll still get `pageCount: 1`.